### ISSUE TITLE
Loss stagnates after first episode on random dataset

### ISSUE THREAD

**Issue by lucasfbn** [tensorflow/tensorflow#56867]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.9.1

### Custom Code

No

### Python version

3.9

#### TL:DR

Loss stagnated after first episode on real-world dataset. Tried same model on a random dataset. Loss stagnated as well. Seems to be related to kernel size. Why is the network seemingly unable to learn anything past the first episode when the kernel size increases above a certain threshold?

Test code/notebook link below.

[Notebook link](https://colab.research.google.com/drive/1sC7KQJwbTPiQ9ZBWZBCzY3if4bSuRzNq?usp=sharing)

#### Context

I'm trying to solve a real-world problem with a simple 1D CNN implemented in TensorFlow. However, the loss "converged"/stayed constant after the first episode for some reason.

After tinkering around with the actual dataset, I tried training the model with a randomly created dataset. And, low and behold, the loss stagnated at the exact same value as it would with the actual dataset.

This seems to be related to the kernel size. However, I can't really make sense of why it results in the exact same constant loss despite being fed two completely different datasets.

#### Question

What is the intuition behind this? Why is the network seemingly unable to learn anything past the first episode when the kernel size increases above a certain threshold?

Note that I get the exact same loss for a real-world dataset, so the problem isn't related to the fact that the example dataset is made up and random.

Thank you for taking the time.

### Standalone code to reproduce the issue

[Notebook link](https://colab.research.google.com/drive/1sC7KQJwbTPiQ9ZBWZBCzY3if4bSuRzNq?usp=sharing)

```python
import numpy as np
import tensorflow as tf

def network(seq_len, n_features):
    model = tf.keras.models.Sequential()
    model.add(tf.keras.layers.Conv1D(filters=64, kernel_size=128, activation='relu',
                                     input_shape=(seq_len, n_features)))
    model.add(tf.keras.layers.Flatten())
    model.add(tf.keras.layers.Dense(units=1, activation='sigmoid'))
    return model


def make_dataset(n, seq_len, n_features, classes):
    x = np.random.random((n, seq_len, n_features))
    y = np.random.randint(0, classes, (n,))

    return x, y, seq_len, n_features

# create random dataset with
# - 10000 sequences, each of which with 2400 timesteps and one feature
# - two classes (0 and 1)
x, y, seq_len, n_features = make_dataset(n=10000, seq_len=2400, n_features=1, classes=2)

model = network(seq_len, n_features)
model.compile(optimizer=tf.keras.optimizers.Adam(),
                  loss=tf.keras.losses.BinaryCrossentropy())
model.fit(x, y, epochs=10, batch_size=64)
```

### Relevant log output

```shell
Epoch 1/10
157/157 [==============================] - 28s 179ms/step - loss: 0.7059
Epoch 2/10
157/157 [==============================] - 27s 172ms/step - loss: 0.6932
Epoch 3/10
157/157 [==============================] - 26s 168ms/step - loss: 0.6931
Epoch 4/10
157/157 [==============================] - 29s 182ms/step - loss: 0.6931
Epoch 5/10
157/157 [==============================] - 28s 176ms/step - loss: 0.6931
Epoch 6/10
157/157 [==============================] - 28s 180ms/step - loss: 0.6931
Epoch 7/10
157/157 [==============================] - 27s 173ms/step - loss: 0.6931
Epoch 8/10
157/157 [==============================] - 27s 172ms/step - loss: 0.6931
Epoch 9/10
157/157 [==============================] - 27s 172ms/step - loss: 0.6931
Epoch 10/10
157/157 [==============================] - 27s 173ms/step - loss: 0.6931
```
</details>

**Comment by AlbertoSinigaglia** [comment]
In my opinion is more related to the activation function than with the loss... probably all the relus are dead... I can reproduce your issue perfectly, but changing the activation function to `"selu"` every works fine

**Comment by lucasfbn** [comment]
Indeed, this seems to be the reason. Thank you for the hint and the quick response!

Any intuition behind why a greater kernel size causes a gradient update that zeroes out all neurons? I can't get my head around that yet.

**Comment by AlbertoSinigaglia** [comment]
@lucasfbn I believe that at this point we can only speculate over why this might happen... but there is no correct answer... for example, given your code, changing the `kernel_inizializer` to something that is not the default, choice, for example `kernel_initializer="random_normal"` everything works, but if you use `kernel_initializer="he_normal"` the learning is extremely faster, but it might be just a case, who knows

At this point, considering that the problem might just be the initialization and not even the activation function, I don't think that we can do much reasoning without getting in the detail of the initializer

**Comment by tilakrayal** [comment]
@lucasfbn,
Could you please take a look at the [comment](https://github.com/tensorflow/tensorflow/issues/56867#issuecomment-1193146763) provided and let us know if the issue still persists. Thank you!

**Comment by lucasfbn** [comment]
Closing as this appears to be related to deep learning theory and not necessarily to TensorFlow. Thank you for your comments, @AlbertoSinigaglia.

If anyone got an explanation of why this is happening, I'd still appreciate it, as I couldn't find anything in the literature.
